doc/compiler/optimizer/DataFlowEngine.md

```diff
@@ -127,4 +127,65 @@ In general using structures as our program fragments enables us to save roughly
+blocks apart. Local Transparency determines if `x+y` can flow through all the blocks to the optimal spot where it should be computed.
+If the block does not change `x` or `y`, `x+y` can be moved pass this block. If `x+y` is computed earlier, and then it is computed again,
+and `x` and `y` have not been changed in between, the two computations can be commoned by Partial Redundancy Elimination (PRE).
+The value of `x+y` can be stored in a temporary which can be used in later blocks instead of computing `x+y` again.
+
+[TR_LocalTransparency](https://github.com/eclipse/omr/blob/f2bc4f8f6eb09f6cc8fc4ba48717de4880b970e3/compiler/optimizer/LocalAnalysis.hpp#L230)
+runs before [TR_LocalAnticipatability](https://github.com/eclipse/omr/blob/f2bc4f8f6eb09f6cc8fc4ba48717de4880b970e3/compiler/optimizer/LocalAnalysis.hpp#L286):
+- `TR_LocalTransparency::_transparencyInfo[symRefNum]`: A bit vector indicates which expressions (identified by local index) will not be
+affected by the sym ref if the sym ref is changed in the block
+- `TR_LocalTransparency::_info[blockNum]._analysisInfo`: A bit vector indicates whether or not the expression (identified by local index)
+is transparent to the block
+
+### Local Anticipatability
+
+Local Transparency does not look at whether or not an expression is evaluated in a certain block. It looks at an expression regardless
+of where it is computed. Local Anticipatablility looks at the expressions that are computed in the block. If an expression is locally
+anticipatable in a block, the expression is computed in the block and can be moved out of the block without changing the value of the
+expression. For example, if `x+y` is computed in `block_10`, the expression `x+y` can be a candidate of being locally anticipatable in
+`block_10`. If `x+y` is not computed in `block_10`, the expression `x+y` will not be in the Local Anticipatablility set.
+
+[TR_LocalAnticipatability](https://github.com/eclipse/omr/blob/f2bc4f8f6eb09f6cc8fc4ba48717de4880b970e3/compiler/optimizer/LocalAnalysis.hpp#L286)
+runs after [TR_LocalTransparency](https://github.com/eclipse/omr/blob/f2bc4f8f6eb09f6cc8fc4ba48717de4880b970e3/compiler/optimizer/LocalAnalysis.hpp#L230)
+- `TR_LocalAnticipatability::_info[blockNum]._analysisInfo`: A bit vector indicates whether or not the value of an expression
+(identified by local index) would be changed if it were computed at the very start of the block (upward anticipatable notion)
+- `TR_LocalAnticipatability::_info[blockNum]._downwardExposedAnalysisInfo`: A bit vector indicates whether or not the value of an
+expression (identified by local index) would be changed if it were computed at the very end of the block
+- `TR_LocalAnticipatability::_info[blockNum]._downwardExposedStoreAnalysisInfo`: A bit vector indicates whether or not the value of a
+store expression (identified by local index) would be changed if it were computed at the very end of the block
+
```

Comment:
It may be worth mentioning that the upward and downward exposed notions of this analysis are both valid for their own use cases. In an ideal world, we would be doing both kinds of analysis but this takes up more compile time than we can likely afford in a JIT compiler (especially as the following dataflow analyses would also need to solve for both notions). So, we currently only use the downward exposed notion for our implementation since we feel that this is relevant for  more of the common / expected cases to be optimized.

Reply:
Addressed in 3f2681c ([Compare](https://github.com/eclipse/omr/compare/2ce35b20627da2bed865b10fa8f8fc3891568233..3f2681c893aa2af174d12f8e50e06afdedaf7e06)). Ready for another review. Thanks!